### PR TITLE
Fix issue with airflow-2-6-0rc1

### PR DESCRIPTION
**Pull request by pankajastro**
# Description
## What is the current behavior?
Tested with Airflow 2.6rc and found there are a couple of issues related
1. serialization/deserialization - Table and Column
2. Test failing `airflow.jobs.scheduler_job.SchedulerJob` and `airflow.jobs.backfill_job.BackfillJob` Classes removed.

Closes: https://github.com/astronomer/astro-sdk/issues/1904

## What is the new behavior?
Added fixes for the above issues.

## Does this introduce a breaking change?
Nope.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by dimberman — COMMENTED** [review]
Overall LGTM minus a few nits. Can you please add some more description as to why there is Serialization issues with airflow 1.6? Also I think you should change the title to something like "Fix serialization issues related to Airflow 2.6"

File: python-sdk/src/astro/sql/operators/cleanup.py [hunk 1]
```diff
@@ -215,11 +215,11 @@ def _is_single_worker_mode(cls, current_dagrun: DagRun) -> bool:
 
     @staticmethod
     def _get_executor_from_job_id(job_id: int) -> str | None:
-        from airflow.jobs.base_job import BaseJob
+        from airflow.jobs.job import Job
         from airflow.utils.session import create_session
 
         with create_session() as session:
-            job = session.get(BaseJob, job_id)
+            job = session.get(Job, job_id)
```
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply python-sdk/src/astro/sql/operators/cleanup.py:222]: No, it won't we need to add checks based on the version.

File: python-sdk/src/astro/table.py [hunk 2]
```diff
@@ -178,6 +178,24 @@ def openlineage_emit_temp_table_event(self):
             isinstance(self, TempTable) and OPENLINEAGE_EMIT_TEMP_TABLE_EVENT
         )
 
+    def serialize(self) -> dict[str, Any]:
+        return {
+            "name": self.name,
+            "temp": self.temp,
+            "conn_id": self.conn_id,
+            "metadata.schema": self.metadata.schema,
+            "metadata.database": self.metadata.database,
```
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply python-sdk/src/astro/table.py:187]: No reason, I'll make that change

File: python-sdk/src/astro/sql/operators/cleanup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply python-sdk/src/astro/sql/operators/cleanup.py:226]: Done.

File: .github/workflows/ci-python-sdk.yaml [hunk 3]
```diff
@@ -242,7 +242,7 @@ jobs:
           key: ${{ runner.os }}-2.5-${{ hashFiles('python-sdk/pyproject.toml') }}-${{ hashFiles('python-sdk/src/astro/__init__.py') }}
       - run: sqlite3 /tmp/sqlite_default.db "VACUUM;"
       - run: pip3 install nox
-      - run: nox -s "test-${{ matrix.version }}(airflow='2.5.3')" -- tests/ --cov=src --cov-report=xml --cov-branch
+      - run: nox -s "test-${{ matrix.version }}(airflow='2.6.0rc1')" -- tests/ --cov=src --cov-report=xml --cov-branch
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply .github/workflows/ci-python-sdk.yaml:245]: Also, we decide to run test on `2.6.0` then we should update the ci job name to `2-6`

**Review by pankajastro — COMMENTED** [review]
Thank you @utkarsharma2 for looking into it. LGTM.

File: .github/workflows/ci-python-sdk.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply .github/workflows/ci-python-sdk.yaml:245]: yup, in my pipeline. Will do that, once they release 2.6.0

[the remark this answers is not in the document]
Comment (tatiana) [reply .github/workflows/ci-python-sdk.yaml]: I just updated Airflow 2.6 was released on https://pypi.org/project/apache-airflow/2.6.0/

**Review by tatiana — APPROVED** [review]
This is a blocker for the 1.6 release, which is expected by customers urgently. For this reason, I'm approving and merging the PR before we meet our desired code coverage (currently, we only have 76.19% of diff hit, while our target for diff coverage is 86.13%).

File: python-sdk/src/astro/table.py [hunk 4]
```diff
@@ -50,6 +50,7 @@ class BaseTable:
     """
 
     template_fields = ("name",)
+    version: ClassVar[int] = 1
```
Comment (tatiana) [inline python-sdk/src/astro/table.py:53]: I really liked this approach of versioning the serialisable objects!

**Pull request merged**